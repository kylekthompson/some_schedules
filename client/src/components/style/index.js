export { default as Button } from 'src/components/style/button';
export { default as Card } from 'src/components/style/card';
export { default as Container } from 'src/components/style/container';
export { default as Link } from 'src/components/style/link';
export { default as Text } from 'src/components/style/text';
