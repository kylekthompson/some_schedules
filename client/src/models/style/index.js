export { default as colors } from 'src/models/style/colors';
export { default as fonts } from 'src/models/style/fonts';
export { default as shadows } from 'src/models/style/shadows';
