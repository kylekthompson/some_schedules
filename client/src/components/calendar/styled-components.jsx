import styled, { css } from 'styled-components';
import { colors, fonts, links } from 'models/styles';

export const Container = styled.div`
  border-bottom: 1px solid ${colors.lightGrey()};
  display: flex;
  flex-direction: column;
  padding: 20px;
`;

export const DayContainer = styled.div`
  align-items: center;
  border-radius: 13px;
  cursor: pointer;
  display: flex;
  flex: 1;
  font-size: 12px;
  height: 26px;
  justify-content: center;
  margin: 2px;
  width: 26px;

  ${({ selected }) => {
    if (selected) {
      return css`
        background-color: ${colors.shakespeareBlue()};
      `;
    }

    return css`
      background-color: ${colors.catskillWhite()};
    `;
  }} &:hover {
    ${({ selected }) => {
      if (selected) {
        return css`
          background-color: ${colors.shakespeareBlue()};
        `;
      }

      return css`
        background-color: ${colors.lightGrey()};
      `;
    }};
  }
`;

export const DayText = styled.p`
  ${fonts.regular}
  color: ${colors.darkGrey()};
  font-size: 12px;
  ${({ activeMonth }) =>
    !activeMonth
      ? css`
          color: ${colors.darkGrey(0.5)};
        `
      : css``}
  ${({ selected }) =>
    selected
      ? css`
          color: ${colors.white()};
        `
      : css``}
`;

export const HeaderDay = styled.div`
  flex: 1;
  text-align: center;
`;

export const HeaderText = styled.p`
  ${fonts.semiBold}
  color: ${colors.darkGrey(0.5)};
  font-size: 12px;
`;

export const MonthContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const NavigationButton = styled.a`
  ${links.darkLink}
  color: ${colors.darkGrey(0.5)};

  &:not(:first-child) {
    margin-left: 20px;
  }
`;

export const NavigationContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: row;
  margin-bottom: 20px;
`;

export const NavigationText = styled.p`
  ${fonts.semiBold}
  color: ${colors.darkGrey(0.5)};
  font-size: 12px;
  margin-right: auto;
`;

export const Week = styled.div`
  display: flex;
  flex-direction: row;
`;
