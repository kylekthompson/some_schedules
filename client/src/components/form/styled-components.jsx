import styled from 'styled-components';
import { colors, fonts } from 'models/styles';

export const Container = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: 400px) {
    width: 400px;
  }
`;

export const ErrorContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: row;
  margin-top: 4px;
`;

export const ErrorIcon = styled.i.attrs({
  className: 'fal fa-exclamation-circle',
})`
  color: ${colors.stilettoRed()};
  margin-right: 6px;
`;

export const ErrorText = styled.p`
  ${fonts.light}
  color: ${colors.white()};
  font-size: 14px;
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
`;

export const HeaderContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  padding-bottom: 60px;
`;

export const Input = styled.input`
  ${fonts.regular}
  background-color: ${colors.white(0.0)};
  border: none;
  border-bottom: 1px solid ${({ isValid }) => {
    if (isValid) {
      return colors.catskillWhite();
    }

    return colors.stilettoRed();
  }};
  color: ${colors.white()};
  font-size: 14px;
  padding: 5px;

  &:focus {
    border-bottom: 1px solid ${({ isValid }) => {
      if (isValid) {
        return colors.bahamaBlue();
      }

      return colors.stilettoRed();
    }};
    outline: none;
  }

  &::placeholder {
    color: ${colors.white(0.5)};
  }
`;

export const Label = styled.p`
  ${fonts.light}
  color: ${colors.white()};
  font-size: 14px;

  &:not(:first-child) {
    margin-top: 30px;
  }
`;

export const Submit = styled.input.attrs({
  type: 'submit',
})`
  ${fonts.regular}
  background-color: ${colors.white(0.0)};
  border: 1px solid ${colors.white(0.25)};
  border-radius: 2px;
  box-shadow: none;
  color: ${colors.white()};
  font-size: 14px;
  margin-top: 30px;
  padding: 5px;

  &:focus,
  &:hover {
    background-color: ${colors.white(0.01)};
    box-shadow: 1px 1px 4px ${colors.black(0.2)};
    cursor: pointer;
    outline: none;
  }

  &:active {
    box-shadow: 3px 3px 12px ${colors.black(0.2)};
    outline: none;
  }

  &:disabled {
    background-color: ${colors.white(0.05)};
    border: none;
    box-shadow: none;
    color: ${colors.white(0.25)};
    cursor: auto;
    outline: none;
  }

  &::placeholder {
    color: ${colors.white(0.5)};
  }
`;

export const Subtitle = styled.p`
  ${fonts.light}
  color: ${colors.white()};
  font-size: 14px;
`;

export const Title = styled.p`
  ${fonts.light}
  color: ${colors.white()};
  font-size: 40px;
  margin-bottom: 10px;
`;
