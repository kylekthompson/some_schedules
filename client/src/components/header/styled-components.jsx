import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { fonts, layout, links } from 'models/styles';
import { header } from 'models/constants';

export const Container = styled.div`
  align-items: center;
  display: flex;
  flex: none;
  flex-direction: row;
  height: ${layout.headerHeight};
`;

export const HeaderLink = styled(Link)`
  ${fonts.regular} font-size: 14px;

  &:not(:first-child) {
    margin-left: 20px;
  }

  ${({ theme }) => {
    if (theme === header.DARK_THEME) {
      return links.darkLink;
    }

    return links.lightLink;
  }};
`;

export const LogoLink = styled(HeaderLink)`
  ${fonts.extraBold} flex: 0;
  font-size: 22px;
`;
