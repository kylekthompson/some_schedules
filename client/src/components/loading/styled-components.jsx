import styled from 'styled-components';
import { fonts } from 'models/styles';

export const Container = styled.div`
  align-items: center;
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: center;
  height: 100%;
`;

export const Message = styled.p`
  ${fonts.regular} font-size: 12px;
`;
