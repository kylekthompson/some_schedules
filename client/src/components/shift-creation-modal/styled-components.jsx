import styled, { css } from 'styled-components';
import { colors, fonts } from 'models/styles';

export const BackgroundMuter = styled.div`
  background-color: rgba(0, 0, 0, 0.2);
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 2000;
`;

export const CreateButton = styled.button`
  background-color: #fd4b00;
  border: 1px solid #fd4b00;
  border-radius: 4px;
  color: white;
  cursor: pointer;
  line-height: 1;
  margin-left: 4px;
  padding: 7px;
  text-decoration: none;

  ${({ disabled }) => {
    if (disabled) {
      return css`
        border-color: lightgrey;
        background-color: lightgrey;
      `;
    }

    return '';
  }};
`;

export const DismissButton = styled.a.attrs({
  children: 'Ã—',
})`
  font-size: 18px;
  float: right;
  cursor: pointer;
  text-decoration: none;
  color: #333;

  &:hover {
    color: #333;
    text-decoration: none;
  }
`;

export const Modal = styled.div`
  background-color: white;
  border-radius: 4px;
  box-shadow: 4px 4px 4px 1px rgba(0, 0, 0, 0.1);
  padding: 10px;
  position: absolute;
  z-index: 2001;

  &:before {
    border-style: solid;
    border-width: 10px;
    content: '';
    display: block;
    position: absolute;

    ${({ x }) => {
      if (x > document.body.clientWidth / 2) {
        return css`
          right: 10px;
        `;
      }

      return css`
        left: 10px;
      `;
    }} ${({ y }) => {
      if (y > document.body.clientHeight / 2) {
        return css`
          border-color: white transparent transparent transparent;
          bottom: -20px;
        `;
      }

      return css`
        border-color: transparent transparent white transparent;
        top: -20px;
      `;
    }};
  }

  ${({ x }) => {
    if (x > document.body.clientWidth / 2) {
      return css`
        right: ${document.body.clientWidth - x - 20}px;
      `;
    }

    return css`
      left: ${x - 20}px;
    `;
  }} ${({ y }) => {
    if (y > document.body.clientHeight / 2) {
      return css`
        bottom: ${document.body.clientHeight - y + 10}px;
      `;
    }

    return css`
      top: ${y + 10}px;
    `;
  }};
`;

export const Separator = styled.hr`
  margin: 10px -10px;
`;

export const Text = styled.span`
  ${fonts.regular} font-size: 18px;
`;

export const TimeInput = styled.input`
  border: 1px solid ${colors.lightGrey()};
  border-radius: 4px;
  line-height: 1;
  padding: 6px;

  &:focus {
    outline: none;
  }
`;
