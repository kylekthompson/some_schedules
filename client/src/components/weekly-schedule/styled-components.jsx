import styled from 'styled-components';
import { colors, fonts } from 'models/styles';

export const Cell = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  padding: 8px;

  &:not(:last-child) {
    border-right: 1px solid ${colors.lightGrey()};
  }
`;

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
`;

export const HeaderCell = styled(Cell)`
  padding: 8px;
`;

export const EmptyHeaderCell = styled(HeaderCell)`
  min-width: 150px;
  max-width: 150px;
`;

export const NameCell = styled(Cell)`
  height: 60px;
  justify-content: center;
  max-width: 150px;
  min-width: 150px;
`;

export const RowContainer = styled.div`
  border-bottom: 1px solid ${colors.lightGrey()};
  display: flex;
  flex-direction: row;
`;

export const Scroller = styled.div`
  overflow-y: scroll;
  overflow-x: hidden;
`;

export const ShiftContainer = styled.div`
  align-items: center;
  border-radius: 2px;
  display: flex;
  flex: none;
  flex-direction: row;
  margin-bottom: 5px;

  :last-child {
    margin-bottom: 0;
  }
`;

export const WeekdayText = styled.p`
  ${fonts.regular} font-size: 12px;
`;

export const DateText = styled.p`
  ${fonts.semiBold} font-size: 48px;
  line-height: 36px;
`;

export const HeaderDayContainer = styled.div`
  flex: 1;
  flex-direction: column;
`;

export const HeaderContainer = styled.div`
  border-bottom: 1px solid ${colors.lightGrey()};
  display: flex;
  flex: none;
  flex-direction: row;
`;

export const ColoredDot = styled.div`
  background-color: ${({ color }) => color};
  border-radius: 10px;
  height: 10px;
  margin-right: 5px;
  width: 10px;
`;

export const PublishedIcon = styled.i.attrs({
  className: ({ isPublished }) =>
    isPublished ? 'fal fa-eye' : 'fal fa-eye-slash',
})`
  margin-right: 5px;
`;

export const ShiftText = styled.p`
  ${fonts.regular} font-size: 12px;
  margin: 0;
`;
