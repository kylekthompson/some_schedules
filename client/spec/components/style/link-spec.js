import React from 'react';
import { Link } from 'src/components/style';
import { css } from 'styled-components';
import { mountAsApp } from 'spec/support/mount';

describe('<Link />', () => {
  describe('without any props', () => {
    it('is an anchor with padding', () => {
      const { container } = mountAsApp(<Link to="/foo" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
