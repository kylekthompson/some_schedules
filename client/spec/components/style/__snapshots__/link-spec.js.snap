// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> without any props is an anchor with padding 1`] = `
.c0 {
  padding: 5px 15px;
}

<a
  class="c0"
  href="/foo"
/>
`;
